### PR TITLE
Fixed B105: email confirmation back button.

### PR DESCRIPTION
**Pull request by madalinazanficu**
As email-confirmation screen is similar with ChangePasswordConfirmation, I have removed the email-confirmation screen, and added a new prop to ChangePasswordConfirmation.

### CODE REVIEW

File: mobile/app/(app)/change-password.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ const ChangePassword = () => {
   };
 
   if (successfullyChanged) {
-    return <ChangePasswordConfirmation />;
+    return <ChangePasswordConfirmation emailConfirmation={false} />;
```
Comment (luciatugui) [inline mobile/app/(app)/change-password.tsx:92]: given that we are using the ChangePasswordConfirmation component both for confirming a password change or for the email sent step in forgot password, maybe we should also pass as a prop the action text on the back button (when we're in the forgot password case, the button should display `Back to log in`, whereas on confirming the password change we need `Back to app` )

Comment (madalinazanficu) [reply mobile/app/(app)/change-password.tsx:92]: discussed offline

**Comment by madalinazanficu** [comment]
For testing:
- use charlie@example.com, for validating the change password confirmation view
- use: user@example.com, for validating the email error view.

File: mobile/app/forgot-password.tsx [hunk 2]
```diff
@@ -3,34 +3,50 @@ import { Screen } from "../components/Screen";
...
-
-  // TODO: Implement the onSubmit function
-  const onSubmit = (data: FormData) => {
-    console.log("Forgot password for email: ", data.email);
-    router.push("/email-confirmation");
+  // Submit handler - forgot password
+  const onSubmit = async (data: FormData) => {
+    try {
+      const payload: ForgotPasswwordPayload = { email: data.email };
+      await forgotPassword(payload);
```
Comment (luciatugui) [inline mobile/app/forgot-password.tsx:38]: we should use a mutation for this, not call the api function directly, please look up the other examples with useMutation

File: mobile/app/forgot-password.tsx [hunk 3]
```diff
@@ -3,34 +3,50 @@ import { Screen } from "../components/Screen";
...
 };
 
 const ForgotPassword = () => {
+  const insets = useSafeAreaInsets();
+  const { t } = useTranslation("reset");
+  const [emailConfirmation, setEmailConfirmation] = useState(false);
+  const [authError, setAuthError] = useState(false);
+
+  // React Hook form
   const { handleSubmit, control, formState } = useForm<FormData>({
```
Comment (luciatugui) [inline mobile/app/forgot-password.tsx:29]: for using the errors we can take them directly from formState, no need to use formState twice

Comment (madalinazanficu) [reply mobile/app/forgot-password.tsx]: done

**Review by luciatugui — APPROVED** [review]

**Pull request merged**